### ISSUE TITLE
Shortcodes not being rendered inside IF/ENDIF code

### ISSUE THREAD

**Issue by Frisoni** [pods-framework/pods#6290]
## Description

Inside a Template I use some shortcodes from my theme.
After 2.8.2 update, shortcodes inside IF/ELSE/ENDIF are not converting to html.

Started this topic on https://wordpress.org/support/topic/shortcode-problem-inside-if-endif/#post-15039635

Example:

```
[if adotado]
[button text="Adotado" color="alert" radius="10" expand="true" icon="icon-heart" icon_pos="left"]
[else]
[button text="Saiba Mais" radius="10" expand="true" icon="icon-heart" icon_pos="left" link="{@permalink}"]
[/if]
```
Render this code:

```
<a href="https://woocommerce-494822-2148013.cloudwaysapps.com/viramor/denis/" target="_self" class="button primary expand"  style="border-radius:10px;">
  <i class="icon-heart" ></i>  <span>Saiba Mais</span>
  </a>
[button text="Saiba Mais" radius="10" expand="true" icon="icon-heart" icon_pos="left" link="https://woocommerce-494822-2148013.cloudwaysapps.com/viramor/denis/"]

```
## Testing instructions

Inside a normal page I call this shortcode:
`[pods name="diva" template="Diva - Item - Loop" orderby="post_date DESC" limit="-1"]
`
In this template I have the following code:

`
[if produto]
[button text="{@produto}" letter_case="lowercase" style="underline" size="small" icon="icon-star" icon_pos="left" link="{@produto.permalink}"]
[else]
[button text="{@nome_produto}" letter_case="lowercase" style="underline" size="small" icon="icon-star" icon_pos="left"]
[/if]
{@produto} and other variables are corrected translated, but the button shortcode don’t. So, I ended up with this string in my page:
[button text="Peruca Wig - ELENA" letter_case="lowercase" style="underline" size="small" icon="icon-star" icon_pos="left" link="https://woocommerce-511254-2145904.cloudwaysapps.com/perucas-e-lace-wigs/peruca-wig-elena/"]

`
This button shortcode is from my theme, Flatsome.

HTML shows [button] shortcode, as drescribed first.

## Expected behavior

That button shortcode render as html

## Pods version

2.8.3

## Other super helpful information

I think I found the difference between 2.7.31 and 2.8.3.
In file **components\Templates\includes\functions-view_template.php** on **frontier_if_block** function

Version 2.7.3 returns:
`return do_shortcode( $template );`
Which correctly render shortcodes to HTML

Version 2.8 starts to return this:
`return frontier_do_shortcode( $template );`
And inside frontier_do_shortcode there is no call to do_shortcode to correctly render.

Best regards,
Frisoni

**Comment by JoryHogeveen** [comment]
@Frisoni 
Did you add `define( 'PODS_SHORTCODE_ALLOW_SUB_SHORTCODES', true );` to your wp config file?

**Comment by Frisoni** [comment]
Hi @JoryHogeveen .

Yes, I have PODS_SHORTCODE_ALLOW_SUB_SHORTCODES = true and this code:
```
add_filter( 'pods_shortcode', function( $tags )  {
  $tags[ 'shortcodes' ] = true;
    return $tags;
  });
```

**Comment by sc0ttkclark** [comment]
Can you provide a package of your configuration (Pods Admin > Migrate Packages) using the Migrate Packages component (Pods Admin > Components)? That might help us to see if we can reproduce the issue.

Also helpful would be access to your Site Health information (Tools > Site Health > Info Tab > Copy site info to clipboard) would also help us determine if there are any potential conflicts from other plugins or your theme.

**Comment by Frisoni** [comment]
Hi @sc0ttkclark.

I created a brand new WP installation for tests, What I found is that this is an incompatibility with enclosing shortcodes created and used by Flatsome theme. Maybe with other themes using shortcodes, but I didn't test any other.

You can check this wp install in the following link: https://wordpress-494822-2269592.cloudwaysapps.com/wp-admin/
User/pass: admin/admin

I created a book pod, with a Single item template.
Flatsome has [col_inner][/col_inner] shortcode, so in this case it will not reder shortcode inside IF on template:

```
[col_inner]
[if book_year]
IF: 
[test_shortcode]
[/if]
[/col_inner]
```

You can check directly on book link: https://wordpress-494822-2269592.cloudwaysapps.com/index.php/book/the-if-book/

I created [test_shortcode] just to return "Shortcode Test!", for testing.

Hope this helps.

**Comment by sc0ttkclark** [comment]
Fixed via 941e2d11e05e26520d600464ce97aaa589ce09c7

The way to utilize this functionality is by using the new constant `PODS_TEMPLATES_ALLOW_OTHER_SHORTCODES` via:

`define( 'PODS_TEMPLATES_ALLOW_OTHER_SHORTCODES', true );`
